### PR TITLE
Feat: Add Vitest

### PR DESCRIPTION
**Pull request by KirillDogadin-std**
closes https://github.com/makerdao-ses/switchboard-boilerplate/issues/11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,11 @@ npm run dev
 # Typechecking (via TypeScript / tsc)
 npm run typecheck
 
-# Linting (via eslint)
```
Comment (BracketJohn) [inline README.md:23]: what happened to eslint doc? 😢

File: vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  test: {
+    coverage: {
+      provider: "istanbul",
+      reporter: ["text", "json", "html"],
```
Comment (BracketJohn) [inline vitest.config.ts:7]: what does this mean? If its the kind of file formatas that are supported: why do we not need to specify js / ts?

**Review by BracketJohn — CHANGES_REQUESTED** [review]
Please also:
- [ ] add a ci step that runs tests
- [ ] enforce coverage of 90 percent (if possible at the moment)

File: README.md (hunk 1 above)
Comment (KirillDogadin-std) [reply README.md:23]: i thought the command was removed but someone forgot to remove the comment as well

File: vitest.config.ts (hunk 2 above)
Comment (KirillDogadin-std) [reply vitest.config.ts:7]:
```
npx vitest run --coverage --reporter=json
```

to see the json formatted output. but seems to work without this.

File: README.md [hunk 3]
```diff
@@ -21,6 +21,13 @@ npm run dev
 npm run typecheck
 
 # Linting (via eslint)
+npm run lint
+
+# Testing via vitest
+npm run test
+
+# Coverage via vitest, istambul coverage provider
+npm run test:cov
```
Comment (BracketJohn) [inline README.md:30]: let's turn on coverage by default!

**Review by BracketJohn — COMMENTED** [review]
you ignored my general review comment from last time

File: README.md (hunk 3 above)
Comment (BracketJohn) [reply README.md]: is coverage enforced by default now? I thought to rememvber that one has to turn in on

**Comment by BracketJohn** [comment]
test cov is not working: Right now it shows 0 LoC, but this is not true -> it should find the src/* code and detect that 0 percent of it are covered.

**Review by BracketJohn — APPROVED** [review]

**Pull request merged**